hbase-server/src/main/java/org/apache/hadoop/hbase/master/zksyncer/ClientZKSyncer.java

```diff
@@ -145,8 +177,7 @@ private final void setDataForClientZkUntilSuccess(String node, byte[] data)
           LOG.warn(
             "Failed to create znode " + node + " due to: " + e.getMessage() + ", will retry later");
         }
-      } catch (KeeperException.ConnectionLossException
-          | KeeperException.SessionExpiredException ee) {
+      } catch (KeeperException.SessionExpiredException see) {
```

Comment:
So we choose not to reconnect if there are some issue on network stability that cause zookeeper connection loss? And it seems the `reconnectAfterExpiration` is not following the same handling on line 173-174

Reply:
ConnectionLoss just means the connection is broken, the zk library will reconnect automatically to resume the session if possible.